wom/src/main/scala/wom/types/WomCompositeType.scala

```diff
@@ -0,0 +1,53 @@
+package wom.types
+
+import cats.instances.list._
+import cats.syntax.traverse._
+import cats.syntax.validated._
+import common.validation.ErrorOr.ErrorOr
+import common.validation.Validation._
+import spray.json.JsObject
+import wom.values.{WomMap, WomObject, WomObjectLike, WomValue}
+
+case class WomCompositeType(typeMap: Map[String, WomType]) extends WomObjectTypeLike {
```

Comment:
This should be subject to intense seagulling to make sure it's not stupid (especially @cjllanwarne )

Reply:
When regular seagulling just won't do.